web/db/actions/Post.ts

```diff
@@ -82,6 +82,26 @@ async function getResizedUploadUrl(uuid: string): Promise<string> {
   return `${consts.baseUrl}/api/resizedUpload/${token}`;
 }
 
+async function deleteAttachments(keysToDelete: string[]) {
+  //will make key splitting more efficient later
+  const objectsToDelete = keysToDelete.map((keyToDelete) => ({
+    Key:
+      keyToDelete.split("/")[keyToDelete.split("/").length - 2] +
+      "/" +
+      keyToDelete.split("/")[keyToDelete.split("/").length - 1],
+  }));
```

Comment:
So keyToDelete will look like this `https://pub-27cb50e8f369479a8f312921bab07d36.r2.dev/65024a562a5311729da58f7b/schedule%20(4).png`- `https://bucketurl/postid/imagename`. I just wanted the `postid/imagename` part of it because that's the key in the bucket. I just used that code when I was just working on the function. I will go in later to make it more efficient.

Reply:
Ah, got it. In that case, you can modify the `getPost` method to take in another boolean parameter `publicAttachmentUrls`. Change the logic so that only when `publicAttachmentUrls` is true, the modification on `post.attachments` is done. The keys themselves are stored in the database, so you won't need the key splitting logic after this change.

Reply:
Make sure to update all uses of `getPost` after you make this change!